src/workerd/io/worker-interface.c++

```diff
@@ -76,21 +76,21 @@ public:
 
   kj::Promise<CustomEvent::Result> customEvent(kj::Own<CustomEvent> event) override {
     KJ_IF_SOME(w, worker) {
-      co_return co_await w.get()->customEvent(kj::mv(event));
+      co_return co_await w->customEvent(kj::mv(event));
     } else {
       co_await promise;
       co_return co_await KJ_ASSERT_NONNULL(worker)->customEvent(kj::mv(event));
     }
   }
 
 private:
-  kj::ForkedPromise<void> promise;
+  kj::Promise<void> promise;
```

Comment:
This seems unsafe if more than one method is called?

It's often the case that a `WorkerInterface` object is used to deliver only one event, but not always.

Reply:
Well even if it's called multiple times it shouldn't be called simultaneously from multiple threads which means that the first execution will still resolve the promise and set worker after which all subsequent executions will not try to await the promise.
This isn't really all too different to the original solution as we didn't addBranch on every infocation.

Reply:
The only usage for this ForkedPromise was that we had branched in the case of prewarm because that handling was done in a waitUntilTask and by the time it ran the promise might have already been resolved.
I removed that part of the code when switching prewarm to return a promise so it's no longer required.

Reply:
You could get two overlapping calls in a single thread. These returns promises after all -- the caller could call another method without first waiting for the first promise to complete.

> This isn't really all too different to the original solution as we didn't addBranch on every invocation.

`co_await forkedPromise` implicitly waits on a branch, hence doesn't consume the promise. `co_await promise`, however, consumes the promise.

Reply:
I see, makes sense now, I'll revert this change, thanks for the explanation 😄